.github/workflows/ci-test.yml

```diff
@@ -125,6 +144,314 @@ jobs:
+          docker login --username AWS --password-stdin "${AWS_ECR_ACCOUNT_URL}"
+          docker/docker_republish.sh -t pre_${BRANCH}_${GIT_REV} -o land_${GIT_REV} -r ${AWS_ECR_ACCOUNT_URL} -d
+
+  land-blocking-test:
+    name: Run land blocking test
+    runs-on: self-hosted
+    needs: build-images
+    # NOTE the total time should cover build (~10 min) and test (~20 min).
+    # The additional time can cover the retries and wait.
+    timeout-minutes: 60
+    steps:
+      - uses: actions/checkout@v1
+      - name: Get PR number
+        id: pr-num
+        uses: actions/github-script@v3
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          result-encoding: string
+          script: |
+            // Find the number of the pull request that trigggers this push
+            let pr_num = 0;
+            let commit_message = context.payload.head_commit.message;
+            let re = /.*[^]Closes:\s\#(\d+)$/;
+            if (re.test(commit_message)) {
+              let match = re.exec(commit_message);
+              pr_num = match[1];
+              return pr_num
+            } else {
+              console.log("GH event payload\n", context.payload);
+              throw "Did not find pull request num in commit message. -\\_(O_o)_/-"
+            }
+      - name: Save PR number
+        run: |
+          echo "PR_NUM=${{steps.pr-num.outputs.result}}" >> $GITHUB_ENV
+      - name: Get PR base ref
+        id: pr-base-ref
+        uses: actions/github-script@v3
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          result-encoding: string
+          script: |
+            let pr_data = await github.pulls.get({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: process.env.PR_NUM
+            });
+            try {
+              let base_ref = pr_data.data.base.ref;
+              console.log("Base ref:", base_ref);
+              if (!base_ref) {
+                throw "Could not find base ref"
+              }
+              return base_ref
+            } catch (err) {
+              console.log("GH PR payload\n", pr_data);
+              throw err
+            }
+      - name: Setup env
+        run: |
+          BASE_REF=${{steps.pr-base-ref.outputs.result}}
+          HEAD_GIT_REV=$(git rev-parse --short=8 HEAD)
+          echo "BASE_REF=$BASE_REF" >> $GITHUB_ENV
+          echo "HEAD_GIT_REV=$HEAD_GIT_REV" >> $GITHUB_ENV
+          echo "TEST_TAG=land_$HEAD_GIT_REV" >> $GITHUB_ENV
+          echo "BASE_GIT_REV=$(git rev-parse --short=8 origin/$BASE_REF)" >> $GITHUB_ENV
+      - name: Check kill switch
+        id: check_ks
+        run: |
+          if ${{ secrets.KILL_SWITCH_LAND_BLOCKING_TEST }} || ! .circleci/should_run_lbt.sh ; then
+            echo "::set-output name=should_run::false";
+          else
+            echo "::set-output name=should_run::true";
+          fi;
+      - name: Determine which cluster-test suite to run
```

Comment:
Is it possible to make the test setup and execution steps into its own actions (like build-setup, build-teardown etc in https://github.com/diem/diem/tree/master/.github/actions)?